### PR TITLE
Feature/async response

### PR DESCRIPTION
**Pull request by akinjide**
- Implement Promises and Callback fn (legacy fallback interface).
- Install Q library for test.
- Update test spec.
- Update README for Promise and Callback usage

✌️

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,39 +1,36 @@
...
  * Validate phone number
  * @param  {string}  n
- * @return {Boolean} 
+ * @return {Boolean}
  */
-function isValid(n) {
-
-  var firstChar;
+function isValid(n, callback) {
+  if ('function' != typeof callback) callback = function () {};
```
Comment (dokasto) [inline index.js]: This is currently not backward compatible & will break existing code.

I'd check if `callback` argument was passed in the first place & use promises then.

**Review by dokasto — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (akinjide) [reply index.js]: Better still we update the `package.json` version to `2.0.2`, stating that it's not backward compatible, or include a `CHANGELOG.md` showing the changes.

Comment (akinjide) [reply index.js]: Btw, `"I'd check if callback argument was passed in the first place & use promises then."`, this was implemented 

```
16 | if ('function' != typeof callback) callback = function () {};
34 | promise.then(callback.bind(null, null), callback);
``` 
resolving with either a promise or callback.

**Comment by akinjide** [comment]
Added commit that resolves issues with backward compatibility

**Comment by akinjide** [comment]
You might need to disable travis cache to pass test.

**Pull request closed without merging**